client/app/states/services/custom_button_details/custom_button_details.state.js

```diff
@@ -25,6 +25,28 @@ function getStates () {
         dialog: resolveDialog,
         service: resolveService
       }
+    },
```

Comment:
these two states appear very similar, instead of duplicating the code with a mild name change, why not make the vmId an optional param on the existing state?

Reply:
Thought about it.  It is really just a judgement call.  Initially I wanted to make it this way so we could refresh the page successfully but it looks like we are passing button data and that isn't reproducible so we can't refresh either way.  I am kinda fine either way.